app/field.py

```diff
@@ -17,20 +17,37 @@ def __init__(self, master, settings):
         self.number_of_bombs.set(settings.number_of_bombs)
         self.bomb_counter = tk.Label(self, textvariable=self.number_of_bombs, font="courier 15 bold", fg="red")
         self.bomb_counter.grid(column=0, columnspan=len(self.columns), row=0)
+        self.folder = os.path.dirname(__file__)
+        self.cell_images = {
+            "closed": tk.PhotoImage(file=(os.path.join(self.folder, "../img/closed.png"))),
+            "last": tk.PhotoImage(file=(os.path.join(self.folder, "../img/boom.png"))),
+            "bomb": tk.PhotoImage(file=(os.path.join(self.folder, "../img/bomb.png"))),
+            "not_bomb": tk.PhotoImage(file=(os.path.join(self.folder, "../img/wrong.png"))),
+            "marked": tk.PhotoImage(file=(os.path.join(self.folder, "../img/marked.png"))),
+            "0": tk.PhotoImage(file=(os.path.join(self.folder, "../img/empty.png"))),
+            "1": tk.PhotoImage(file=(os.path.join(self.folder, "../img/1.png"))),
+            "2": tk.PhotoImage(file=(os.path.join(self.folder, "../img/2.png"))),
+            "3": tk.PhotoImage(file=(os.path.join(self.folder, "../img/3.png"))),
+            "4": tk.PhotoImage(file=(os.path.join(self.folder, "../img/4.png"))),
+            "5": tk.PhotoImage(file=(os.path.join(self.folder, "../img/5.png"))),
+            "6": tk.PhotoImage(file=(os.path.join(self.folder, "../img/6.png"))),
+            "7": tk.PhotoImage(file=(os.path.join(self.folder, "../img/7.png"))),
+            "8": tk.PhotoImage(file=(os.path.join(self.folder, "../img/8.png")))
+        }

```

Comment:
You still create new Image objects when the game is restarted. It would be better to store images at the level of Application and avoid re-creating them after the game is restarted.

Having said that, the current approach does ensure that stale Image objects are at some point destroyed by the garbage collector, so one might argue that it is not a leak. I think it is good enough.